### PR TITLE
Fix bug: localtime is not thread-safe,then changed to localtime_r.

### PR DESCRIPTION
**Pull request by worker24h**
Fix bug: function localtime is not thread-safe,then changed to localtime_r.
Issue： https://github.com/apache/incubator-doris/issues/1615

### CODE REVIEW

File: be/src/exec/base_scanner.cpp [hunk 1]
```diff
@@ -144,12 +145,21 @@ bool BaseScanner::fill_dest_tuple(const Slice& line, Tuple* dest_tuple, MemPool*
...
-            if (_strict_mode && (_src_slot_descs_order_by_dest[dest_index] != nullptr)
-                && !_src_tuple->is_null(_src_slot_descs_order_by_dest[dest_index]->null_indicator_offset())) {
+            SlotDescriptor* slotDescriptor = _src_slot_descs_order_by_dest[dest_index];
+            if (_strict_mode && (slotDescriptor != nullptr)&& !_src_tuple->is_null(slotDescriptor->null_indicator_offset())) {
+                StringValue* rawValue = _src_tuple->get_string_slot(slotDescriptor->tuple_offset());
                 std::stringstream error_msg;
-                error_msg << "column(" << slot_desc->col_name() << ") value is incorrect "
-                    << "while strict mode is " << std::boolalpha << _strict_mode;
-                _state->append_error_msg_to_file("", error_msg.str());
+                if (rawValue) {//is not null then get raw value
```
Comment (imay) [inline be/src/exec/base_scanner.cpp]:
```suggestion
                if (rawValue != nullptr) {//is not null then get raw value
```

File: be/src/exec/base_scanner.cpp [hunk 2]
```diff
@@ -144,12 +145,21 @@ bool BaseScanner::fill_dest_tuple(const Slice& line, Tuple* dest_tuple, MemPool*
...
+                    std::string rawString = rawValue->to_string();
+                    error_msg << " column(" << slot_desc->col_name() << ") value("<< rawString <<") is incorrect "
+                        << "while strict mode is " << std::boolalpha << _strict_mode;
+                    _state->append_error_msg_to_file(rawString, error_msg.str());
+                } else {
+                    error_msg << " column(" << slot_desc->col_name() << ") value is incorrect "
+                        << "while strict mode is " << std::boolalpha << _strict_mode;
+                    _state->append_error_msg_to_file("", error_msg.str());
+                }
+                LOG(WARNING) << error_msg.str();
```
Comment (imay) [inline be/src/exec/base_scanner.cpp]: remove this log, we should not print error log for wrong data when load. This is a normal case.

File: be/src/exec/base_scanner.cpp [hunk 3]
```diff
@@ -144,12 +145,21 @@ bool BaseScanner::fill_dest_tuple(const Slice& line, Tuple* dest_tuple, MemPool*
...
+                if (rawValue) {//is not null then get raw value
+                    std::string rawString = rawValue->to_string();
+                    error_msg << " column(" << slot_desc->col_name() << ") value("<< rawString <<") is incorrect "
+                        << "while strict mode is " << std::boolalpha << _strict_mode;
+                    _state->append_error_msg_to_file(rawString, error_msg.str());
+                } else {
+                    error_msg << " column(" << slot_desc->col_name() << ") value is incorrect "
+                        << "while strict mode is " << std::boolalpha << _strict_mode;
+                    _state->append_error_msg_to_file("", error_msg.str());
+                }
```
Comment (imay) [inline be/src/exec/base_scanner.cpp]: and I think you can reuse the append string logic.

std::string raw_string;
if (raw_value != nullptr) {
    raw_string = raw_value->to_string();
}
                    error_msg << " column(" << slot_desc->col_name() << ") value is incorrect "
                        << "while strict mode is " << std::boolalpha << _strict_mode;
                    _state->append_error_msg_to_file("", error_msg.str());

File: be/src/exec/base_scanner.cpp [hunk 4]
```diff
@@ -144,12 +145,21 @@ bool BaseScanner::fill_dest_tuple(const Slice& line, Tuple* dest_tuple, MemPool*
         ExprContext* ctx = _dest_expr_ctx[dest_index];
         void* value = ctx->get_value(_src_tuple_row);
         if (value == nullptr) {
-            if (_strict_mode && (_src_slot_descs_order_by_dest[dest_index] != nullptr)
-                && !_src_tuple->is_null(_src_slot_descs_order_by_dest[dest_index]->null_indicator_offset())) {
+            SlotDescriptor* slotDescriptor = _src_slot_descs_order_by_dest[dest_index];
+            if (_strict_mode && (slotDescriptor != nullptr)&& !_src_tuple->is_null(slotDescriptor->null_indicator_offset())) {
+                StringValue* rawValue = _src_tuple->get_string_slot(slotDescriptor->tuple_offset());
```
Comment (imay) [inline be/src/exec/base_scanner.cpp]:
```suggestion
                StringValue* raw_value = _src_tuple->get_string_slot(slotDescriptor->tuple_offset());
```

File: be/src/exec/base_scanner.cpp [hunk 5]
```diff
@@ -144,12 +145,21 @@ bool BaseScanner::fill_dest_tuple(const Slice& line, Tuple* dest_tuple, MemPool*
         ExprContext* ctx = _dest_expr_ctx[dest_index];
         void* value = ctx->get_value(_src_tuple_row);
         if (value == nullptr) {
-            if (_strict_mode && (_src_slot_descs_order_by_dest[dest_index] != nullptr)
-                && !_src_tuple->is_null(_src_slot_descs_order_by_dest[dest_index]->null_indicator_offset())) {
+            SlotDescriptor* slotDescriptor = _src_slot_descs_order_by_dest[dest_index];
```
Comment (imay) [inline be/src/exec/base_scanner.cpp]:
```suggestion
            SlotDescriptor* slot_descriptor = _src_slot_descs_order_by_dest[dest_index];
```

**Review by imay — CHANGES_REQUESTED** [review]

File: be/src/exec/base_scanner.cpp (hunk 5 above)
Comment (worker24h) [reply be/src/exec/base_scanner.cpp]: OK

File: be/src/exec/base_scanner.cpp (hunk 4 above)
Comment (worker24h) [reply be/src/exec/base_scanner.cpp]: ok

File: be/src/exec/base_scanner.cpp (hunk 3 above)
Comment (worker24h) [reply be/src/exec/base_scanner.cpp]: ok

File: be/src/exec/base_scanner.cpp (hunk 2 above)
Comment (worker24h) [reply be/src/exec/base_scanner.cpp]: ok

File: be/src/exec/base_scanner.cpp (hunk 1 above)
Comment (worker24h) [reply be/src/exec/base_scanner.cpp]: ok

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**